docs/contracts/staking-router.md

```diff
@@ -3,9 +3,843 @@
+
+StakingRouter performs a number of functions, including:
+- a registry of staking modules,
+- allocating stake to modules, and
+- distributing protocol fees.
+
+## Module Management
+
+### Registering a module
+
+Modules are registered with StakingRouter through the Lido DAO voting process. To be considered by the governance, the applying module contract should implement the appropriate module interface, meet security requirements, and have a fee structure aligned with the Lido protocol sustainability. Once voted in, the module starts receiving stake and protocol fees.
+
+Staking modules are registered using the `addStakingModule` function, providing details such as:
+- the module's name: a human-readable name;
+- address of the deployed staking module contract;
+- target share, a relative hard cap on deposits within Lido;
+- module fee, a percentage of staking rewards to be awarded to the module,
+- and treasury fee, a percentage of staking rewards to be directed to the protocol treasury.
+
+### Pausing modules
+
+Each staking module has a status: a state that determines whether the module can perform deposits and receive rewards:
+- `Active`, can make deposits and receives rewards,
+- `DepositsPaused`, deposits are not allowed but receives rewards,
+- `Stopped`, cannot make deposits and does not receive rewards.
+
+```solidity
+enum StakingModuleStatus {
+	Active, // deposits and rewards allowed
+	DepositsPaused, // deposits NOT allowed, rewards allowed
+	Stopped // deposits and rewards NOT allowed
+}
+```
+
+### Exited and stuck validators
+
+When the withdrawal requests exceed the Lido reserved buffer, the protocol signals to node operators to start exiting validators to cover the withdrawals. In this connection, StakingRouter distinguishes two types of validator states:
+- exited validators,
+- and stuck validators, meaning those validators that failed to comply with the exit signal.
+
+StakingRouter keeps track of both of these types of validators in order to correctly allocate stake and penalize stuck validators. These stats are kept up to date via the oracles submitting the data to the contract.
+
+## Stake allocation
+
+StakingRouter carries out a vital task of distributing depositable ether to staking modules in a manner that aligns with their growth targets set by the DAO. This design ensures a regulated and controlled growth for the modules that have been newly integrated into the system. The principles governing this methodology are comprehensively discussed in [ADR: Staking Router](https://hackmd.io/f1wvHzpjTIq41-GCrdaMjw?view#Target-shares).
+
+### Deposit
+
+The deposit workflow involves submitting batches of 32 ether deposits, along with associated validator keys, to DepositContract in one transaction. Given that each staking module handles its own deposits, every batch deposit is restricted to keys originating from a single module.
```

Comment:
Please add a link to DepositContract
https://ethereum.org/en/staking/deposit-contract/

Reply:
would be great to use the surrounding ticks